proposals/compatibility-mode.md

```diff
@@ -16,17 +16,35 @@ Since WebGPU Compatibility mode is a subset of WebGPU, all valid Compatibility m
 
 ## WebGPU Spec Changes
 
-When calling `GPU.requestAdapter()`, passing `featureLevel = "compatibility"` in the `GPURequestAdapterOptions` will indicate to the User Agent to select the Compatibility subset of WebGPU. Any Devices created from the resulting Adapter on supporting UAs will support only Compatibility mode. Calls to APIs unsupported by Compatibility mode will result in validation errors.
+### Adapters
 
-Note that a supporting User Agent may return a `featureLevel = "compatibility"` Adapter which is backed by a fully WebGPU-capable hardware adapter, such as D3D12, Metal or Vulkan, so long as it validates all subsequent API calls made on the Adapter and the objects it vends against the Compatibility subset.
+When calling `GPU.requestAdapter()`, passing `featureLevel: "compatibility"` in the `GPURequestAdapterOptions` will indicate to the User Agent a request to select the Compatibility subset of WebGPU. If the request is honored by the UA, the resulting adapter will be a "Compatibility-defaulting" Adapter. When calling `adapter.requestDevice()` on a Compatibility-defaulting Adapter, the default set of capabilities (features and limits) is more restrictive than on a Core-defaulting Adapter.
+
+A Compatibility-defaulting Adapter may be able to create devices with more capabilities than the default Compatibility subset (as requested via `requiredFeatures`/`requiredLimits`).
+
+A Compatibility-defaulting Adapter may even be "Core-capable"; if it is, it lists `"webgpu-core"` as a supported feature.
+Requesting `requiredFeatures: ["webgpu-core"]` increases all capabilities to at least the Core defaults (ignoring any limit requests which are worse than Core).
```

Comment:
I wonder if we're going to see developers using this pattern with the following code:

```js
const adapter = await navigator.gpu.requestAdapter({ featureLevel: "compatibility" });

const requiredFeatures = [];
if (adapter.features.has("webgpu-core")) {
  // If the Compatibility-defaulting Adapter is "Core-capable", let's increase
  // all the capabilities to at least the Core defaults.
  requiredFeatures.push("webgpu-core");
}

const device = await adapter.requestDevice({ requiredFeatures });
```

Reply:
I expect some devs will do that.  Some devs will also request all features and limits

```js
const adapter = await navigator.gpu.requestAdapter(...);
const device = await adapter.requestDevice({
  requiredFeatures: adapter.features,
  requiredLimits: objLikeToObj(adapter.limits),
});
```

Then it's up to them to gate usage on `device.features.has('some-feature')` and `device.limits.xxx`

That's okay right? I mean it's not ideal because they should test that their code works locally with the min limits and least features but I suspect lots of code will try to use the maximum features/limits.

Reply:
That's exactly the intended way to use it, except you would also enable some optional features of your app when core is available. I had an example in the old PR that I didn't carry over into this one:

https://github.com/gpuweb/gpuweb/pull/4898/files#diff-5e8d5eb23a52df345b7c87a865b36211b7b7b8f555ef3e4270357ee735ee2a53R77-R91

Reply:
It'd be nice to have this javascript snippet as well then in this proposal to help developers understand about how the `"webgpu-core"` feature works with compatibility mode.

Reply:
Added an example, slightly more detailed than before too.